src/databricks/labs/remorph/cli.py

```diff
@@ -76,20 +81,71 @@ def transpile(
     """[EXPERIMENTAL] reconciles source to databricks datasets"""
     logger.info(f"user: {w.current_user.me()}")
 
-    if source.lower() not in {"databricks", "snowflake", "oracle"}:
-        raise_validation_exception(
-            f"Error: Invalid value for '--source': '{source}' is not one of 'databricks', 'snowflake', 'oracle'. "
-        )
-    if report.lower() not in {"data", "schema", "all", "row"}:
-        raise_validation_exception(
-            f"Error: Invalid value for '--report': '{report}' is not one of 'data', 'schema', 'all' , 'row'"
-        )
+    installation = Installation.assume_user_home(w, "remorph")
+    _prompts = Prompts()
+
+    reconcile_config = None
+    try:
+        logger.info("Loading ReconcileConfig `reconcile.yml` from Databricks Workspace...")
+        reconcile_config = installation.load(ReconcileConfig)
+    except NotFound as err:
+        logger.warning(f"Cannot find previous `reconcile` installation: {err}")
+    except (PermissionDenied, SerdeError, ValueError, AttributeError):
+        logger.warning(f"Existing installation at {installation.install_folder()} is corrupted. Skipping...")
+
+    reconfigure_msg = "Please use `remorph install` to re-configure ** reconcile ** module"
+    # Re-configure `reconcile` module:
+    # * when there is no `reconcile.yml` config on Databricks workspace OR
+    # * when there is a `reconcile.yml` config and user wants to overwrite it
+    if not reconcile_config:
+        logger.error(f"`reconcile_config` not found / corrupted on Databricks Workspace.\n{reconfigure_msg}")
+        return
+    if _prompts.confirm(
+        f"Would you like to overwrite workspace `reconcile_config` values:\n" f" {reconcile_config.__dict__}?"
+    ):
+        logger.info(reconfigure_msg)
+        return
+
+    catalog_or_schema = (
+        reconcile_config.config.source_catalog
+        if reconcile_config.config.source_catalog
+        else reconcile_config.config.source_schema
+    )
+
+    # Creates the filename in the format of : `recon_config_<SOURCE>_<CATALOG_OR_SCHEMA>_<FILTER_TYPE>.json`
+    # Ex: recon_config_snowflake_sample_data_all.json
+    filename = f"recon_config_{reconcile_config.data_source}_{catalog_or_schema}_{reconcile_config.report_type}.json"
+
+    try:
+        logger.info(f"Loading TableRecon `{filename}` from Databricks Workspace...")
+        table_recon = installation.load(type_ref=TableRecon, filename=filename)
+    except NotFound as err:
+        logger.error(f"Cannot find previous `reconcile` installation: {err}")
+        raise err
+    except (PermissionDenied, SerdeError, ValueError, AttributeError) as ex:
+        logger.error(f"Existing installation at {installation.install_folder()}/{filename} is corrupted. Skipping...")
+        raise ex
+
+    assert table_recon, (
+        f"Error: Cannot load `recon_config` from {installation.install_folder()}/{filename}. "
+        f"Please re-generate using `remorph generate-recon-config` "
```

Comment:
Remove the second line add it once generate config cli is available

Reply:
Good catch.